src/common/cuepoint/cuepoint-manager.js

```diff
@@ -0,0 +1,87 @@
+//@flow
+import {Cue, FakeEvent, TextTrack, EventManager} from '@playkit-js/playkit-js';
+import {CUE_POINTS_TEXT_TRACK, CUE_POINT_KEY} from './cuepoint-type';
+import {CuePointEventType} from './cuepoint-event-type';
+
+interface CuePoint {
+  id: string;
+  startTime: number;
+  endTime: number;
+}
+
+export class CuePointManager {
+  _player: KalturaPlayer;
+  _eventManager: EventManager;
+  _textTrack: TextTrack | null = null;
+
+  constructor(player: KalturaPlayer) {
+    this._player = player;
+    this._eventManager = new EventManager();
+    this._eventManager.listen(player, player.Event.Core.PLAYER_RESET, () => this.reset());
```

Comment:
not need. call to CuePointManager.reset on kaltura-player reset. like https://github.com/kaltura/kaltura-player-js/blob/master/src/kaltura-player.js#L320